### PR TITLE
Add linter rules and format check to CI

### PR DESCRIPTION
**Pull request by lsizemore8**
### Summary 

This pull request is the first step towards adding a fully functional linter for Carriage Driver. The ruleset used is pedantic, the ruleset used at Google and is considered a recommended list by Flutter developers. Making this a WIP PR to test if flutter analyze will run during the PR creation. 

- [x] implemented linter
- [ ] fixed Y

### Test Plan 

Ran Flutter analyze and received a list of info warnings, as expected. Moreover, in VS Code, the system does show squiggly marks underneath the bad code practices, as expected. Will do a full test by running the app to ensure no breaking changes from the current fixes soon. 

Note: WIP due to the lack of screenshot-based tests. Only creating the PR to test the system with the CI rules. 

### Notes 
This PR will create a roadblock for future dev work at first, as PRs will not be allowed to merged in unless a developer has no issues from running `flutter analyze`. However, this PR fixes most of the pre-existing issues, so this should not be too big of a problem. 

There is potentially one breaking change: the rule unawaited futures. Due to the usage of pushReplacement, which returns a future, to do navigation rather than a push/pop system, I said the pushReplacement should be unawaited while adding it for the other providers. This will have to be heavily tested, but it should not break for now.

### CODE REVIEW

**Comment by mg2239** [comment]
Updated this PR to use flutter-lints instead of pedantic, which is now deprecated. I also added a format check to the CI, so all future PRs must be formatted using `dart format .` for consistency. Finally I added support for Flutter v2, which required changing some deprecated Button widgets to their new versions. The buttons should be the only actual UI change in this PR.

**Comment by mg2239** [comment]
actually getting some issues with the UI being cutoff + buttons not rendering correctly on iPhone SE, so not quite ready yet

File: lib/pages/Login.dart [hunk 1]
```diff
@@ -17,14 +16,11 @@ class Login extends StatelessWidget {
     return Scaffold(
       backgroundColor: Colors.black,
       body: Container(
```
Comment (helenxuyang) [inline lib/pages/Login.dart:19]: Need to wrap this in a SafeArea or it might put stuff under the notification bar at the top

image

**Review by helenxuyang — APPROVED** [review]
Skimmed through files (most are just whitespace changes) and checked through commits for actual changes, tested on my phone. All looks good to me because formatting changes are just padding/spacing--only one comment about the login page needing a SafeArea. Thanks Matt for the hard work!!

**Pull request merged**